### ISSUE TITLE
Lots of deprecation warnings on Classes using inheritance

### ISSUE THREAD

**Issue by arggh** [jagi/meteor-astronomy#541]
I just updated Astronomy to 2.3.4.

Now all my Astronomy Classes that use inheritance cause this to be logged multiple times upon initialization:

`8dfbc53….js?meteor_js_resource=true:244 Deprecation warning: Usage of the "_isNew" property have been deprecated. Please use the "MyClass.isNew(doc)" function.`

It complains only on Classes using inheritance. I did text-search-check each-and-every package involved, none of them were using `_isNew` (directly, at least) and all of them were updated to use `2.3.4`.

Any ideas?

<details>
```
deprecated	@	deprecated.js:14
get	@	beforeInit.js:16
hasPath	@	jagi_astronomy.js?hash=9d9711a…:15589
has	@	jagi_astronomy.js?hash=9d9711a…:22587
beforeInit	@	beforeInit.js:12
(anonymous function)	@	dispatch_event.js:41
arrayEvery	@	jagi_astronomy.js?hash=9d9711a…:10036
every	@	jagi_astronomy.js?hash=9d9711a…:18590
dispatchEvent	@	dispatch_event.js:40
Class	@	class.js:84
Class	@	class.js:148
Class	@	class.js:174
castToClass	@	castToClass.js:16
(anonymous function)	@	transformToClass.js:24
(anonymous function)	@	wrapTransform.js:9
(anonymous function)	@	minimongo.js?hash=66cc6ab…:1197
Tracker.nonreactive	@	tracker.js?hash=9f8a0ce…:631
wrapped	@	minimongo.js?hash=66cc6ab…:1196
(anonymous function)	@	minimongo.js?hash=66cc6ab…:210
_.each._.forEach	@	underscore.js?hash=cde485f…:149
LocalCollection.Cursor.forEach	@	minimongo.js?hash=66cc6ab…:205
LocalCollection.Cursor.fetch	@	minimongo.js?hash=66cc6ab…:248
LocalCollection.findOne	@	minimongo.js?hash=66cc6ab…:175
findOne	@	collection.js:337
(anonymous function)	@	find.js:87
MY_CUSTOM_FUNCTION	@	MY_COMPONENT.js:7
getValue	@	manuel_viewmodel.js?hash=d3cf07e…:940
(anonymous function)	@	manuel_viewmodel.js?hash=d3cf07e…:960
(anonymous function)	@	viewmodel-parseBind.coffee:17
fun	@	manuel_viewmodel.js?hash=d3cf07e…:569
(anonymous function)	@	blaze.js?hash=983d07a…:1875
Template._withTemplateInstanceFunc	@	blaze.js?hash=983d07a…:3687
(anonymous function)	@	blaze.js?hash=983d07a…:1873
Blaze._withCurrentView	@	blaze.js?hash=983d07a…:2214
Template.invite_toolbox_menu:anonymous	@	blaze.js?hash=983d07a…:1872
Tracker.Computation._compute	@	tracker.js?hash=9f8a0ce…:339
Tracker.Computation	@	tracker.js?hash=9f8a0ce…:229
Tracker.autorun	@	tracker.js?hash=9f8a0ce…:604
Blaze.View.autorun	@	blaze.js?hash=983d07a…:1885
Blaze.TemplateInstance.autorun	@	blaze.js?hash=983d07a…:3547
autorun	@	manuel_viewmodel.js?hash=d3cf07e…:574
fn	@	viewmodel-parseBind.coffee:17
bind	@	viewmodel-parseBind.coffee:17
ViewModel.bindSingle	@	manuel_viewmodel.js?hash=d3cf07e…:598
ViewModel.bind	@	manuel_viewmodel.js?hash=d3cf07e…:1107
(anonymous function)	@	manuel_viewmodel.js?hash=d3cf07e…:297
Tracker._runFlush	@	tracker.js?hash=9f8a0ce…:539
onGlobalMessage	@	meteor.js?hash=e3f53db…:401
```
</details>

**Comment by lukejagodzinski** [comment]
What does it mean:

> none of them were using _isNew (directly, at least)

Not directly, that means? `doc._isNew` is deprecated at all so it should not be used. However, if you want to disable deprecation warnings you can use set the `Astro.config.verbose = false` config flag. Anyway, I would recommend replacing all occurrences of `_isNew`.
Deprecation warnings may also be printed when using softremove behavior. I'm preparing fix for it, that will not use `_isNew` anymore.

**Comment by arggh** [comment]
> What does it mean:
> 
> none of them were using _isNew (directly, at least)

I meant, that maybe some Astronomy behavior or Astronomy's internal module is still using `_isNew`, since a text search in all of my projects gave 0 results. I'm not using it _anywhere_.

I have my Classes all wrapped in the same package, and they are quite identical in terms of structure, yet the ones using inheritance **all** throw this deprecation warning, so I thought maybe it's a bug.

Edit: Also, I'm not using softremove behavior

**Comment by lukejagodzinski** [comment]
Ok I will test inheritance if it's causing this problem. Maybe it does. Will let you know.

**Comment by lukejagodzinski** [comment]
In deed there is some problem with usage of the `_isNew` on child creation. I will fix it an publish bugfix version.

**Comment by lukejagodzinski** [comment]
Fixed in 2.3.5
